File: pfe-2018-network-journey-simulator/src/models/eventAttendancePeak.go
package models

import (
	"time"
)

type EventAttendancePeak struct {
	time		time.Time
	idStation	int
	size		int
	finished	bool
}

func NewEventAttendancePeak(idStation int, time time.Time, size int) EventAttendancePeak {
	return EventAttendancePeak {
		time :			time,
		idStation : 	idStation,
		size : 			size,
		finished :		false,
	}
}

func (eap *EventAttendancePeak) Finished() bool {
	return eap.finished
}

func (eap *EventAttendancePeak) SetFinished(f bool) {
	eap.finished = f
}

func (eap *EventAttendancePeak) Time() time.Time {
	return eap.time
}

func (eap *EventAttendancePeak) IdStation() int {
	return eap.idStation
}

func (eap *EventAttendancePeak) SetidStation(id int) {
	eap.idStation = id
}

func (eap *EventAttendancePeak) Size() int {
	return eap.size
}

func (eap *EventAttendancePeak) SetSize(s int) {
	eap.size = s
}

File: railway-editor_java_version/Railway-editor/src/model/EventAttendancePeak.java
package model;

/**
 * Model class extending event which describes an attendance peak on a station. 
 * @author arthu
 *
 */
public class EventAttendancePeak extends Event {

	private int idStation;
	private int size;
	
	/**Constructor.
	 * @param startTime event startTime
	 * @param endTime event endTime
	 * @param type eventType
	 */
	public EventAttendancePeak(String startTime, String endTime, EventType type) {
		super(startTime, endTime, type);
		setEventName("attendancePeak");
	}

	/**get the id of the station concerned by the peak.
	 * @return int id
	 */
	public int getIdStation() {
		return idStation;
	}

	/**set the id of the station concerned.
	 * @param idStation id station concerned by the peak
	 */
	public void setIdStation(int idStation) {
		this.idStation = idStation;
	}

	/** get the peak size.
	 * @return int size
	 */
	public int getSize() {
		return size;
	}

	/** set the peak size.
	 * @param size size of the peak
	 */
	public void setSize(int size) {
		this.size = size;
	}
	
	
}
